Update functions.qmd (#1178)
hadley/r4ds: functions.qmd

  rng <- range(x, na.rm = TRUE, finite = TRUE)
  (x - rng[1]) / (rng[2] - rng[1])
}

> 

Author: Throughout I tried to format the code so that it looks the same, i.e. adding a space whenever a new function has been defined. Please ignore if it this is unnecessary. It's just something I noticed.

rescale01(x)
```

These changes illustrate an important benefit of functions: because we've moved the repeated code into a function, we only need to make the change in one place.

### Mutate functions

Now you've got the basic idea of functions, let's take a look at a whole bunch of examples.

> We'll start by looking at "mutate" functions, i.e. functions that work like `mutate()` and `filter()` because they return an output of the same length as the input.

Author: Maybe I am not understanding it correctly (it's late already). How do `filter()` and `mutate()` return an output of the same length as the input? Mutate can add new columns to the tibble and filter often reduces the amount of rows.

Reviewer: I think this should be "work well _inside_ of `mutate()` and `filter()`"

> We'll start by looking at "mutate" functions, i.e. functions that work like `mutate()` and `filter()` because they return an output of the same length as the input.

Reviewer: ```suggestion
We'll start by looking at "mutate" functions, i.e. functions that work well inside of `mutate()` and `filter()` because they return an output of the same length as the input.
```

### Summary functions

> Another important family of vector functions are summary functions, functions that return a single value for use in `summarize()`.

Reviewer 2: ```suggestion
Another important family of vector functions is summary functions, functions that return a single value for use in `summarize()`.
```

      fun = fun,
    )
}

diamonds |> hex_plot(carat, price, depth)
```

### Combining with dplyr

Some of the most useful helpers combine a dash of dplyr with ggplot2.
For example, if you might want to do a vertical bar chart where you automatically sort the bars in frequency order using `fct_infreq()`.
Since the bar chart is vertical, we also need to reverse the usual order to get the highest values at the top (also note the `:=` operator, which allows you to inject names with glue syntax on the left-hand side of ⁠`:=`;⁠ type: ?\`:=\` for more details):

> 

Author: This is just a suggestion but the ":=" is new (I haven't seen it myself either before). So perhaps pointing it out may be a good idea? My edit here is just a suggestion. Doesn't have to be that way.

Reviewer: I'll flesh this out a bit more after merging.

    ggplot(aes(y = {{ var }})) + 
    geom_bar()
}

diamonds |> sorted_bars(cut)
```

> Or you maybe you want to make it easy to draw a bar plot just for a subset of the data:

Reviewer 2: ```suggestion
Or maybe you want to make it easy to draw a bar plot just for a subset of the data:
```
